symengine/polynomial.cpp

```diff
@@ -159,6 +156,23 @@ integer_class UnivariateIntPolynomial::eval_bit(const int &x) const {
     return ans;
 }
 
+integer_class UnivariateIntPolynomial::horner_scheme(map_uint_mpz::const_iterator curr, unsigned int &l, const integer_class &x) const {
+
+    integer_class ans(1), temp;
+
+    if (curr == dict_.end())
+        return integer_class(0);
+
+    while ((*curr).first - l > 0) {
+        l++;
+        ans = ans * x;
+    }
+
+    temp = (*curr).second;
+    curr++;
+    return ans * (temp + horner_scheme(curr, l, x));
```

Comment:
Can you turn this into a non-recursive function? For very large polynomials, recursion is not good.